File: internal/model/order.go
package model

type Order struct {
	ID       string
	Quantity int
	Packs    map[int]int
}

type OrderPack struct {
	Pack     Pack
	Quantity int
}

func MakeOrder(orderQuantity int, packs map[int]int) Order {
	if len(packs) == 0 {
		packs = map[int]int{}
	}

	order := Order{
		Quantity: orderQuantity,
		Packs:    packs,
	}

	return order
}

func OrderPackToPackSlice(storedPacks []Pack, orderPacks map[int]int) []OrderPack {
	result := make([]OrderPack, 0, len(orderPacks))
	for amount, quantity := range orderPacks {
		for _, pack := range storedPacks {
			if pack.Amount == amount {
				result = append(result, OrderPack{Pack: pack, Quantity: quantity})
			}
		}
	}
	return result
}

File: ui/pack-management/src/service/order.ts
import { getClientInstance } from './http_client'
import { Pack } from './pack'


export type OrderPack = {
  pack: Pack
  quantity: number
}

export const calculateOrder = async (amount: number): Promise<OrderPack[]> => {
  const client = getClientInstance();
  const response = await client.post(`/order/calculate`, {
    amount,
  })
  if (response.status !== 200) {
    throw new Error('Failed to calculate order', response.data);
  }

  const data = response.data
  
  return data['order_packs'];
}